view/frontend/web/js/action/select-shipping-method-mixin.js

```diff
@@ -0,0 +1,125 @@
+                            'shipping_address': {
+                                'email': shippingEmail,
+                                'telephone': shippingTelephone,
+                                'firstname': shippingFirstname,
+                                'lastname': shippingLastname,
+                                'street': shippingStreet,
+                                'city': shippingCity,
+                                'region': shippingRegion,
+                                'region_id': shippingRegionId,
+                                'region_code': shippingRegionCode,
+                                'country_id': shippingCountryId,
+                                'postcode': shippingPostcode,
+                                'same_as_billing': 0,
+                                'customer_address_id': 0,
+                                'save_in_address_book': 0
+                            },
+                            'billing_address': {
+                                'email': response.billing_address.email,
+                                'telephone': response.billing_address.telephone,
+                                'firstname': response.billing_address.firstname,
+                                'lastname': response.billing_address.lastname,
+                                'street': response.billing_address.street,
+                                'city': response.billing_address.city,
+                                'region': response.billing_address.region,
+                                'region_id': response.billing_address.region_id,
+                                'region_code': response.billing_address.region_code,
+                                'country_id': response.billing_address.country_id,
+                                'postcode': response.billing_address.postcode,
+                                'same_as_billing': 0,
+                                'customer_address_id': 0,
+                                'save_in_address_book': 0
+                            },
+                            'shipping_method_code': shippingMethod['method_code'],
+                            'shipping_carrier_code': shippingMethod['carrier_code']
+                        }
+                    };
+
+                    // update quote_address table with selected shipping method
+                    setShippingInformation(payload, false).then(function () {
+                        customerData.set(shippingMethodChangedEvent, shippingMethod['method_code']);
+                    });
+                });
+
+                // update checkout data with selected shipping method
+                let result = _super(shippingMethod);
+
+                if (shippingMethod) {
+                    // update quote data
+                    window.checkoutConfig.quoteData.base_grand_total = parseFloat(window.checkoutConfig.totalsData.subtotal) + parseFloat(shippingMethod['amount']);
```

Comment:
@peterojo here is where I am setting the base_grand_total to be later used when sending the order amount while mounting the second amazon component in button.js